pandas/core/internals/blocks.py

```diff
@@ -870,7 +870,13 @@ def _replace_coerce(
         """
         if mask.any():
             if not regex:
-                nb = self.coerce_to_target_dtype(value)
+                if isinstance(
+                    self.values, (IntegerArray, FloatingArray)
+                ) and self._can_hold_element(value):
```

Comment:
This seems like a very specific check, can you comment on why it is necessary / makes sense? Also, how does BooleanArray fit in here? Do we have tests on if replacement for BooleanArray works as expected?

Reply:
I'd guess the same underlying problem for nullable dtypes is also responsible for https://github.com/pandas-dev/pandas/issues/40732#issuecomment-816495128, so it would be great if the patch were able to apply more generally and a fix a case like that as well (unless you think the underlying issues are different).

Reply:
I focused the problem encounters in IntegerArray and FloatingArray that's why I added the check. But, i think i should focus on nullable arrays. 
I think for the replace method, we don't have test that cover nullable types. Will try to add some.